### PR TITLE
Remove deprecated OpenApi3 factory

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### CODE REVIEW

**Comment by scholzj** [comment]
@sknot-rh Any plans with this? Or should it be closed?

**Comment by sknot-rh** [comment]
> @sknot-rh Any plans with this? Or should it be closed?

This was blocked by a typo in the spec. It is already fixed.

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 1]
```diff
@@ -648,7 +648,7 @@ void sendToOneStringPartitionTest(VertxTestContext context) throws InterruptedEx
 
         producerService()
             .sendRecordsToPartitionRequest(topic, partition, root, BridgeContentType.KAFKA_JSON_JSON)
-            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: partitionid - Value is not a valid number"));
+            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: partitionid - [Bad Request] Parsing error for parameter partitionid in location PATH: java.lang.NumberFormatException: For input string: \"" + partition + "\""));
```
Comment (PaulRMellor) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:651]:
```suggestion
            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: partitionid - [Bad Request] Parsing error for the partitionid parameter in location PATH: java.lang.NumberFormatException: For input string: \"" + partition + "\""));
```

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerIT.java [hunk 2]
```diff
@@ -464,7 +464,7 @@ void createConsumerWithWrongFetchMinBytes(VertxTestContext context) throws Inter
     @Test
     void createConsumerWithNotExistingParameter(VertxTestContext context) throws InterruptedException, TimeoutException, ExecutionException {
         checkCreatingConsumer("foo", "bar", HttpResponseStatus.BAD_REQUEST,
-                "Validation error on: body - $.foo: is not defined in the schema and the schema does not allow additional properties", context);
+                "Validation error on:  - provided object should not contain additional properties", context);
```
Comment (PaulRMellor) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerIT.java:467]:
```suggestion
                "Validation error on:  - provided object must not contain additional properties", context);
```

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 3]
```diff
@@ -671,8 +671,7 @@ void sendToBothPartitionTest(VertxTestContext context) throws InterruptedExcepti
 
         producerService()
             .sendRecordsToPartitionRequest(topic, partition, root, BridgeContentType.KAFKA_JSON_JSON)
-            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: body.records[0] - " +
-                    "$.records[0].partition: is not defined in the schema and the schema does not allow additional properties"));
+            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object should not contain additional properties"));
```
Comment (PaulRMellor) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:674]:
```suggestion
            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object must not contain additional properties"));
```

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 4]
```diff
@@ -693,7 +692,7 @@ void sendMessageLackingRequiredProperty(VertxTestContext context) throws Throwab
 
         producerService()
             .sendRecordsRequest(topic, root, BridgeContentType.KAFKA_JSON_JSON)
-            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: body.records[0] - $.records[0].value: is missing but it is required"));
+            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object should contain property value"));
```
Comment (PaulRMellor) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:695]:
```suggestion
            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object must contain property value"));
```

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 5]
```diff
@@ -715,8 +714,7 @@ void sendMessageWithUnknownProperty(VertxTestContext context) throws Throwable {
 
         producerService()
             .sendRecordsRequest(topic, root, BridgeContentType.KAFKA_JSON_JSON)
-            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: body.records[0] - " +
-                "$.records[0].foo: is not defined in the schema and the schema does not allow additional properties"));
+            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object should not contain additional properties"));
```
Comment (PaulRMellor) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:717]:
```suggestion
            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object must not contain additional properties"));
```

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 6]
```diff
@@ -818,8 +816,7 @@ void jsonPayloadTest(VertxTestContext context) throws InterruptedException, Exec
 
         producerService()
             .sendRecordsToPartitionRequest(topic, 0, root, BridgeContentType.KAFKA_JSON_JSON)
-            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: body.records[0] - " +
-                "$.records[0].partition: is not defined in the schema and the schema does not allow additional properties"));
+            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object should not contain additional properties"));
```
Comment (PaulRMellor) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:819]:
```suggestion
            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: /records/0 - provided object must not contain additional properties"));
```

**Review by PaulRMellor — APPROVED** [review]

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java (hunk 1 above)
Comment (sknot-rh) [reply src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:651]: These are messages from the vertx library.

**Review by ppatierno — APPROVED** [review]

**Pull request merged**